src/vehicle_move.cpp

```diff
@@ -1129,13 +1248,77 @@ bool vehicle::act_on_map()
         skidding = true;
     }
 
-    // Where do we go
+    bool allow_turn_on_rail = false;
+    if( can_use_rails ) {
```

Comment:
put this all in a separate function, please.

Reply:
done